File: sdk/flutter/test/portal_example.dart
import 'package:flutter/material.dart';
import 'package:levelmoment_ads/levelmoment_ads.dart';

// Call once at app startup.
Future<void> initializeLearning() => LevelMomentAds.instance.initialize();

// Call from the player's learning action. Pass your game's callbacks.
Future<void> showLearningBreak(BuildContext context, VoidCallback grantBonus, VoidCallback resumeGame) async {
  final access = await LevelMomentAds.instance.ensureAccess(context: context, placementId: "YOUR_PLACEMENT_ID");
  if (!context.mounted) return;
  if (access != EnsureSignedInResult.ready) { resumeGame(); return; }
  var earned = false;
  await LevelMomentRewardedAd.load(
    placementId: "YOUR_PLACEMENT_ID",
    adLoadCallback: LevelMomentAdLoadCallback(
      onAdLoaded: (ad) {
        ad.fullScreenContentCallback = LevelMomentFullScreenContentCallback(
          onAdDismissedFullScreenContent: (ad) { ad.dispose(); resumeGame(); },
          onAdFailedToShowFullScreenContent: (ad, error) { ad.dispose(); resumeGame(); },
        );
        ad.show(context: context, onUserEarnedReward: (ad, reward) { if (reward.amount == 1 && !earned) { earned = true; grantBonus(); } });
      },
      onAdFailedToLoad: (error) => resumeGame(),
    ),
  );
}

File: sdk/unity/Samples~/PortalExample.cs
using System;
using LevelMoment;

// Install and configure the supported WebView provider from the Unity guide.
public static class LearningBreaks
{
    // Call once at app startup.
    public static void Initialize() => LevelMomentAds.Initialize(new LevelMomentConfig {

    });

    // Call from the player's learning action. Persist each RewardId in your game
    // before granting so callback retries and webhooks cannot grant twice.
    public static void Show(Action<string> grantBonus, Action resumeGame)
    {
        LevelMomentAds.EnsureAccess("YOUR_PLACEMENT_ID", access => {
            if (access != EnsureSignedInResult.Ready) { resumeGame(); return; }
            bool earned = false;
            RewardedAd.Load("YOUR_PLACEMENT_ID", new RewardedAdLoadCallbacks {
                OnAdLoaded = ad => ad.Show(new RewardedAdShowCallbacks {
                    OnUserEarnedRewardItem = reward => { if (!earned) { earned = true; grantBonus(reward.RewardId); } },
                    OnAdDismissed = () => { ad.Destroy(); resumeGame(); },
                    OnAdFailedToShow = error => { ad.Destroy(); resumeGame(); },
                }),
                OnAdFailedToLoad = error => resumeGame(),
            });
        });
    }
}
